features/utilities/src/main/feature/feature.xml

```diff
@@ -155,4 +159,20 @@
         <bundle>mvn:ddf.platform.io/platform-io-impl/${project.version}</bundle>
     </feature>
 
+    <feature name="jackson" version="${project.version}">
+        <bundle>mvn:com.fasterxml.jackson.core/jackson-core/${jackson.version}</bundle>
+        <bundle>mvn:com.fasterxml.jackson.core/jackson-annotations/${jackson.version}</bundle>
+        <bundle>mvn:com.fasterxml.jackson.core/jackson-databind/${jackson.version}</bundle>
+    </feature>
+
+    <feature name="poi" version="${project.version}">
+        <feature dependency="true">apache-commons</feature>
```

Comment:
❓ Oddly enough, just declaring the `<feature>` dependency without attributes works most of the time so I've never understood why the feature data model needed these attributes. 

I've seen cases where`dependency="true"` is necessary, and also cases where it doesn't help at all. There's also `prerequisite="true"` which I think creates a new OSGi subsystem but sometimes works when `dependency="true"` does not. Have you explored these subtle differences by chance and know precisely what differs between the different strategies?

Reply:
I'm not sure. I didn't actually add it here. This feature is just moved from another file.

In my experience I've not seen it make a difference either. I'm curious too if anybody can explain.

Reply:
I'm not sure it does anything right now. It is just a flag. I believe they have changed behavior in later versions of Karaf to take that flag into account though. I remember seeing a commit or two related to it as I was digging through karaf commits.